test/scripts/integration-tests.sh

```
@@ -379,6 +432,65 @@
+    local test_commands="$2"
+    local expected_responses="$3"
+
+    echo "🔌 Testing WebSocket endpoint: $endpoint..."
+
+    # Check if we have a WebSocket client
+    if [ -z "$WS_COMMAND" ]; then
+        echo "⚠️ No WebSocket client (websocat or wscat) found, skipping WebSocket tests"
+        return 0
+    fi
+
+    local response_file=$(mktemp /tmp/ws_response.XXXXXX)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)